src/main/java/it/cnr/isti/hpc/wikipedia/parser/ArticleParser.java

```diff
@@ -55,24 +62,28 @@
 	private String lang = Language.EN;
 
 	static int shortDescriptionLength = 500;
-	private List<String> redirects;
+	private final List<String> redirects;
+
+	private Pattern patternNE = Pattern.compile(":*([^:]+):(.+)");
+	private Pattern patternNoNameSpace = Pattern.compile(":*([^:]+.*)");
```

Comment:
Even better for these to be static, otherwise they will still be compiled on every `new ArticleParser`.